umbral/keys.py

```diff
@@ -239,3 +261,61 @@ def __eq__(self, other):
         else:
             is_eq = False
         return is_eq
+
+
+class UmbralKeyingMaterial(AbstractUmbralKeyingMaterial):
+    """
+    This class handles keying material for Umbral, by allowing deterministic
+    derivation of UmbralPrivateKeys based on labels. 
+    Don't use this key material directly as a key.
+    
+    """
+
+    def __init__(self, keying_material: bytes=None):
+        """
+        Initializes an UmbralKeyingMaterial.
+        """
+        if keying_material:
+            if len(keying_material) < 32:
+                raise ValueError("UmbralKeyingMaterial must have size at least 32 bytes.")
+            self.keying_material = keying_material
+        else:
+            self.keying_material = os.urandom(32)
+
+    def _get_keying_material(self):
+        return self.keying_material
+
+    def derive_private_key_from_label(self, label: bytes, 
```

Comment:
I think I prefer `derive_privkey_by_label`. We have taken the standard to calling all keys `privkey` or `pubkey` in one word without the underscore.

Reply:
OK, great. I'm all in for reducing verbosity.